### ISSUE TITLE
Optionality of sRGB Images and Depth Images for 3.0 conformance

### ISSUE THREAD

**Issue by bcalidas** [KhronosGroup/OpenCL-CTS#892]
Verify that implementations not supporting either sRGB images or Depth Images (or both) can pass 3.0 conformance and that no additional checks are needed in the image related conformance tests.

**Comment by bashbaug** [comment]
I took a quick glance through the CTS source and I don't think anything needs to change for sRGB images or Depth images - these image formats are only required for OpenCL 2.x devices.

See:

https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_common/harness/imageHelpers.cpp#L3527

We can leave this issue open to confirm, but I believe no action will be required.

**Comment by lakshmih** [comment]
We are reviewing this and running tests to confirm that no changes are needed to CTS.

**Comment by lakshmih** [comment]
Ran following tests all of which test CL_DEPTH and CL_sRGBA formats and they all passed on a 3.0 implementation that doesn't support these:

- cl_copy_images
- cl_fill_images
- cl_read_write_images
- image_streams
- kernel_image_methods
- samplerless_reads
- basic

**Comment by neiltrevett** [comment]
From 15Sep20 call - check if consistency tests are needed for sRGB

**Comment by lakshmih** [comment]
We reviewed this and concluded that no consistency check is needed for the sRGB image format.

However, there is the 2.0 extension cl_khr_srgb_image_writes that adds support for writes to sRGB images so a consistency check can potentially be added to verify that a driver that reports the extension string also exports the corresponding formats.

**Comment by bashbaug** [comment]
> However, there is the 2.0 extension cl_khr_srgb_image_writes that adds support for writes to sRGB images so a consistency check can potentially be added to verify that a driver that reports the extension string also exports the corresponding formats.

Good point - yes, I agree we could add a consistency check for sRGB image writes.  I don't think this is must-have for OpenCL 3.0, but it's probably not too hard to add, either.

A few questions, if we wanted to add a consistency check.  Some of these questions should debatably go to the spec repo vs. this CTS repo:

* Do we want to mandate any specific sRGB image formats for implementations that support `cl_khr_srgb_image_writes`, or just that at least one write-able sRGB image format is supported?  The [spec](https://bashbaug.github.io/OpenCL-Docs/html/OpenCL_Ext.html#cl_khr_srgb_image_writes) doesn't say anything about specific sRGB image formats being required.

* If an implementation supports at least one write-able sRGB image format must it also advertise support for the `cl_khr_srgb_image_writes` extension?  Note that we do not currently mention the `cl_khr_srgb_image_writes` extension anywhere in the API spec other than to generally say that writing to sRGB images is optional.

**Comment by bcalidas** [comment]
Qualcomm can add the consistency checks for sRGB image writes. However, this should not be made blocking for 3.0. 

Regarding the consistency check itself, I don't think the spec requires a minimum set of sRGB images that need to be supported for writes, Therefore we would look for at least 1 writeable sRGB format. We will double check this.

**Comment by bashbaug** [comment]
> However, this should not be made blocking for 3.0.

I agree.  I'll remove the label and take this issue off of the board.  Thanks!

**Comment by rjodinchr** [comment]
Obsolete: Working group discussion and validation confirmed that no CTS changes were needed, as `CL_DEPTH` and `CL_sRGB(A)` image formats are optional in OpenCL 3.0 (mandatory only in OpenCL 2.x), and 3.0 devices without them pass all conformance tests.
